src/Forms/Shared/CategoryListPage.xaml.cs

```diff
@@ -1,40 +1,143 @@
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Esri.ArcGISRuntime.Security;
 using Xamarin.Forms;
 
-namespace ArcGISRuntimeXamarin
+namespace ArcGISRuntime
 {
-    public partial class CategoryListPage : ContentPage
+    public partial class CategoryListPage
     {
-        List<TreeItem> _sampleCategories;
-        public CategoryListPage ()
+        public CategoryListPage()
         {
             Initialize();
-            InitializeComponent ();
+            InitializeComponent();
         }
 
-        async void Initialize()
+        private void Initialize()
         {
-            await SampleManager.Current.InitializeAsync();
-            _sampleCategories= SampleManager.Current.GetSamplesAsTree();
-            BindingContext = _sampleCategories;
+            // Initialize the sample manager.
+            SampleManager.Current.Initialize();
 
+            ViewModel = new SamplesSearchViewModel();
+
+            // Update the binding.
+            BindingContext = ViewModel;
+        }
+
+        private async void OnItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            // Get the selected category.
+            var category = e.Item as SearchableTreeNode;
+            var sample = e.Item as SampleInfo;
+            if (category != null)
+            {
+                // Navigate to the listing page for the category.
+                await Navigation.PushAsync(new SampleListPage(category.Name));
+            }
+            else if (sample != null)
+            {
+                // Navigate to the sample page
+                ShowSample(sample);
+            }
+        }
+
+        private async void ShowSample(SampleInfo item)
+        {
+            try
+            {
+                // Load offline data before showing the sample.
+                if (item.OfflineDataItems != null)
+                {
+                    // Show the wait page.
+                    await Navigation.PushAsync(new WaitPage { Title = item.SampleName }, false);
+
+                    // Wait for the sample data download.
+                    await DataManager.EnsureSampleDataPresent(item);
+
+                    // Remove the waiting page.
+                    await Navigation.PopAsync(false);
+                }
+
+                // Get the sample control from the selected sample.
+                var sampleControl = (ContentPage)SampleManager.Current.SampleToControl(item);
+
+                // Create the sample display page to show the sample and the metadata.
+                SamplePage page = new SamplePage(sampleControl, item);
+
+                // Show the sample.
+                await Navigation.PushAsync(page, true);
+
+                // Clear Credentials
+                foreach (Credential cred in AuthenticationManager.Current.Credentials)
+                {
+                    AuthenticationManager.Current.RemoveCredential(cred);
```

Comment:
Did you mean to clear credentials _before_ or _after_ the sample is displayed? As far as I can tell, this code will run right as the sample is being displayed (i.e. as PushAsync animation is running). I'm not sure if that's intended.

Reply:
Ultimately, I don't think it matters; clearing credentials before sample load or after sample close are functionally the same (sample always runs with cleared credentials). The code could probably be clearer about the intent.